bitriseclient/bitriseclient.go

```diff
@@ -0,0 +1,939 @@
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		return nil
+
+	}); err != nil {
+		return err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+
+	return nil
+}
+
+// ConfirmIdentityUpload ...
+func (client *BitriseClient) ConfirmIdentityUpload(certificateSlug string, certificateUploadName string) error {
+	fmt.Println()
+	log.Infof("Confirm - %s - upload to Bitrise...", certificateUploadName)
+
+	requestURL, err := urlutil.Join(baseURL, appsEndPoint, client.selectedAppSlug, "build-certificates", certificateSlug, "uploaded")
+	if err != nil {
+		return err
+	}
+
+	headers := map[string]string{
+		"Authorization": "token " + client.accessToken,
+	}
+
+	request, err := createRequest("POST", requestURL, headers, map[string]interface{}{})
+	if err != nil {
+		return err
+	}
+
+	// Response struct
+	responseStatusCode := -1
+	requestResponse := ConfirmProvProfileUploadResponse{}
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		// Parse JSON body
+		if err := json.Unmarshal([]byte(body), &requestResponse); err != nil {
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+
+		logDebugPretty(requestResponse)
+
+		return nil
+
+	}); err != nil {
+		return err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+
+	return nil
+}
+
+// -------------------------------------------------
+// -- Commons
+
+func createUploadRequest(requestMethod string, url string, headers map[string]string, files map[string]string) (*http.Request, error) {
+	var fContent []byte
+
+	for _, file := range files {
+		f, err := os.Open(file)
+
+		if err != nil {
+			return nil, err
+		}
+
+		fContent, err = ioutil.ReadAll(f)
+
+		if err != nil {
+			return nil, err
+		}
+
+	}
+
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewReader(fContent))
+
+	if err != nil {
+		return nil, err
+	}
+
+	for key, value := range headers {
+		req.Header.Add(key, value)
+
+	}
+
+	return req, nil
+}
+
+func createRequest(requestMethod string, url string, headers map[string]string, fields map[string]interface{}) (*http.Request, error) {
```

Comment:
createRequest could be method on BitriseClient, since every request needs the auth header it should be handled inside this function, this will shorten the specific requests

Reply:
Not exactly. The  S3 requests don't need headers at all. 
But I have moved the header map to the BitriseClient as an attribute.